# Binary Predictors in a Logistic Regression

Using the same code as in the previous exercise, find the odds of 'duration'. 

What do they tell you?

## Import the relevant libraries

In [50]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
sns.set()
import statsmodels.api as sm
from scipy import stats

## Load the data

Load the ‘Bank_data.csv’ dataset.

In [67]:
data = pd.read_csv('Bank_data.csv')

In [68]:
#Remove redundant index column
data = data.drop(['Unnamed: 0'],axis = 1)

In [69]:
data['y'].unique()

array(['no', 'yes'], dtype=object)

In [70]:
#Create dummy binary variable for the 'y' column
data['y'] = data['y'].map({'no':0,'yes':1})

In [71]:
data['y'].unique()

array([0, 1])

In [72]:
#Update 'may' column to act as binary
data['may'] = data['may'].map({0:0,1:1,2:1,3:1,4:1,5:1})

### Declare the dependent and independent variables

Use 'duration' as the independet variable.

In [73]:
y = data['y']
x1 = data['duration']

### Simple Logistic Regression

Run the regression.

In [75]:
x = sm.add_constant(x1)
reg_log = sm.Logit(y,x)
results_log = reg_log.fit()
results_log.summary()

Optimization terminated successfully.
         Current function value: 0.546118
         Iterations 7


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:                      y   No. Observations:                  518
Model:                          Logit   Df Residuals:                      516
Method:                           MLE   Df Model:                            1
Date:                Mon, 02 Aug 2021   Pseudo R-squ.:                  0.2121
Time:                        11:08:34   Log-Likelihood:                -282.89
converged:                       True   LL-Null:                       -359.05
Covariance Type:            nonrobust   LLR p-value:                 5.387e-35
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const         -1.7001      0.192     -8.863      0.000      -2.076      -1.324
duration       0.0051      0.001      9.159      0.000       0.004       0.006
==============================================================================
"""

### Find the odds of duration

In [78]:
np.exp(0.0051)

1.005113027136717

### The odds increase 0.51% for each increase of 1 in 'duration'